indico/modules/rb/controllers/__init__.py

```diff
@@ -23,3 +25,13 @@ def _check_access(self):
         RHProtected._check_access(self)
         if not rb_check_user_access(session.user):
             raise Forbidden(_('You are not authorized to access the room booking system.'))
+
+    def _validate_reason(self, reason, link):
```

Comment:
This does not belong in `RHRoomBookingBase`. You can add the validation logic in the `CreateBookingSchema`:

```python
    @validates('reason')
    def _check_reason(self, reason, **kwargs):
        if [your checks failed]:
            raise ValidationError(...)
```

Reply:
Actually, I'm not sure if the `@validates` runs when the field has no value (ie it's null). But marking it conditionally required is not trivial, so if that validation function indeed does not work, you may need to use `@validates_schema` which will give you the whole dict of data that was loaded by the schema.

Reply:
Since you need to access the `link_id` I think you'll need to use the schema-level validation in any case. Also double-check if we're actually sending a `link_id` when editing an existing booking!

Reply:
I applied your suggestion.